GVRf/Sample/gvr-vuforia/src/org/gearvrf/vuforiasample/VuforiaSampleScript.java

```diff
@@ -20,157 +13,156 @@
+    private float[] convertMatrix = { 1f, 0f, 0f, 0f, 0f, -1f, 0f, 0f, 0f, 0f,
+            -1f, 0f, 0f, 0f, 0f, 1f };
 
-                    mGVRMVMatrix = mGVRContext.getMainScene()
-                            .getMainCameraRig().getOwnerObject().getTransform()
-                            .getModelMatrix();
+    @Override
+    public void onStep() {
+        if (VuforiaSampleActivity.isVuforiaActive()) {
+            Renderer.getInstance().begin();
+            Renderer.getInstance().bindVideoBackground(0);
+            Renderer.getInstance().end();
+        }
+    }
 
-                    Matrix.multiplyMM(mTotalMVMatrix, 0, mGVRMVMatrix, 0,
-                            mConvertedMVMatrix, 0);
-                    bunny.getTransform().setModelMatrix(mTotalMVMatrix);
-                    break;
-                }
+    public void updateObjectPose(State state) {
+        // did we find any trackables this frame?
+        for (int tIdx = 0; tIdx < state.getNumTrackableResults(); tIdx++) {
+            TrackableResult result = state.getTrackableResult(tIdx);
+            Trackable trackable = result.getTrackable();
+            if (trackable.getId() == 1) {
+                Matrix44F modelViewMatrix_Vuforia = Tool
+                        .convertPose2GLMatrix(result.getPose());
+                mVuforiaMVMatrix = modelViewMatrix_Vuforia.getData();
+
+                Matrix.multiplyMM(mConvertedMVMatrix, 0, convertMatrix, 0,
+                        mVuforiaMVMatrix, 0);
+
+                float scaleFactor = ((ImageTarget) trackable).getSize()
+                        .getData()[0];
+                Matrix.scaleM(mConvertedMVMatrix, 0, scaleFactor, scaleFactor,
+                        scaleFactor);
+
+                mGVRMVMatrix = mGVRContext.getMainScene().getMainCameraRig()
+                        .getTransform().getModelMatrix();
+
+                Matrix.multiplyMM(mTotalMVMatrix, 0, mGVRMVMatrix, 0,
+                        mConvertedMVMatrix, 0);
+                teapot.getTransform().setModelMatrix(mTotalMVMatrix);
+                
+                break;
             }
         }
     }
 
-    private Renderer vuforiaRederer;
+    @SuppressWarnings("unused")
+    private void showMatrix(String name, float[] matrix) {
+        Log.d(TAG, name);
+        Log.d(TAG, String.format("%5.2f %5.2f %5.2f %5.2f", matrix[0],
+                matrix[4], matrix[8], matrix[12]));
+        Log.d(TAG, String.format("%5.2f %5.2f %5.2f %5.2f", matrix[1],
+                matrix[5], matrix[9], matrix[13]));
+        Log.d(TAG, String.format("%5.2f %5.2f %5.2f %5.2f", matrix[2],
+                matrix[6], matrix[10], matrix[14]));
+        Log.d(TAG, String.format("%5.2f %5.2f %5.2f %5.2f", matrix[3],
+                matrix[7], matrix[11], matrix[15]));
+        Log.d(TAG, "\n");
+    }
+
     private float[] mVuforiaMVMatrix;
     private float[] mConvertedMVMatrix;
     private float[] mGVRMVMatrix;
```

Comment:
Can we move these member variables to the top of this code file where others are?